packages/preact-iso/router.js

```diff
@@ -18,6 +20,44 @@ const UPDATE = (state, url, push) => {
 	return url;
 };
 
+const segmentize = (url) => url.replace(/(^\/+|\/+$)/g, '').split('/');
+const exec = (url, route, opts) => {
+	let matches = {},
+		ret;
+
+	url = segmentize(url);
+	route = segmentize(route || '');
```

Comment:
`url` is already normalized now, so I think this can be shrunk to:

```suggestion
const exec = (url, route, opts) => {
	let matches = {},
		ret;

	url = url.slice(1).split('/');
	route = (route || '').slice(1).replace(/\/$/, '').split('/');
```